index.js

```diff
@@ -1,34 +1,53 @@
-import makeTemplate from "lodash.template";
-import { visitParents as visit } from "unist-util-visit-parents";
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @package rehype-lodash-template
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ * @copyright Viktor Yakubiv & Contributors
+ * @license MIT
+ */
 
-const attach = (options = {}) => {
-  const { values = {}, templateSettings } = options;
+import makeTemplate from "lodash.template";
+import { visitParents } from "unist-util-visit-parents";
 
-  const cache = new Map();
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ *
+ * @param {object} param0 { values, templateSettings }
+ * @returns Rehype Plugin
+ */
+export default function attach({ values, templateSettings }) {
+  const valueCache = new Map();
```

Comment:
It was global by purpose. Imaging the following code:
```js
const str = 'hello <%= user %>!'
const template = _.template(str)
const result = compiled({ 'user': 'fred' });
```

Wherever you call `_.template(str)`, it will generate practically the same function for you. If the module imported/loaded/executed more than once but in different places, same template strings will produce same results.

There is no point in local cache. It works in the same way as global one but uses more memory.

---

If you rename, think about name. It is not a value cache, it's a _cache of template functions_ but it is still the only one in the module and there is no point in this extra verbosity.